docs/templating/validations.md

```
@@ -119,6 +119,96 @@ spec:
+Because both `resource` and `parameters` are in scope, a single rule with CEL comprehensions can assert that everything a trait added (e.g. via `forEach` over `parameters`) is still present after the whole trait stack has run — no per-item loop construct is needed:
+
+```yaml
+postRenderValidations:
+  # Assert every volume mount this trait added via forEach over parameters.mounts
+  # is still present after all later traits in the stack have run.
+  - target: { group: apps, version: v1, kind: Deployment }
+    rule: ${parameters.mounts.all(m, resource.spec.template.spec.volumes.exists(v, v.name == m.name))}
+    message: "a later trait removed one of this trait's volume mounts"
+```
+
+Post-render rule failures across all matched resources and all traits are aggregated (joined with `; `) into a single error, mirroring pre-render behavior. A malformed `where` selector is reported immediately as a hard error.
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)